tiledb/sm/query/test/unit_query_condition.cc

```diff
@@ -222,19 +223,18 @@ TEST_CASE(
                   sizeof(int),
                   QueryConditionOp::GT)
               .ok());
-  CHECK(
-      tiledb::test::ast_node_to_str(query_condition1.ast()) ==
-      "y GT 33 33 33 33");
+  check_ast_str(query_condition1, "y GT 33 33 33 33");
 
   QueryCondition combined_and;
   REQUIRE(
       query_condition
           .combine(
               query_condition1, QueryConditionCombinationOp::AND, &combined_and)
           .ok());
-  CHECK(
-      tiledb::test::ast_node_to_str(combined_and.ast()) ==
-      "(x LT 12 ef cd ab AND y GT 33 33 33 33)");
+  check_ast_str(combined_and, "(x LT 12 ef cd ab AND y GT 33 33 33 33)");
```

Comment:
This tests `negate()` but I don't see anything in this file that tests `NOT`.  It doesn't seem like we are actually testing expressions with `NOT` in them, but rather just testing `negative()` on different expressions, meaning what is being tested is only `NOT` at the root of a tree.  I think we need to test more complicated uses of `NOT` than that.

If `NOT` is only going to be applicable from the C API, it needs to be tested there.  

(My own opinion is that `NOT` should not be evaluated only in the C API and that we should be able to build expressions that have `QueryConditionCombinationOp::NOT` in them.)

Reply:
With the current implementation/assumption that NOT can't exist in the AST, there's nothing extra to test. I do need to add C/C++ API tests to prove the the changes in tiledb.cc are correct. The current tests are just covering correctness.